go/vt/sqlparser/ast.go

```diff
@@ -1373,24 +1383,30 @@ func (node *BinaryExpr) WalkSubtree(visit Visit) error {
 
 // UnaryExpr represents a unary value expression.
 type UnaryExpr struct {
-	Operator byte
+	Operator string
 	Expr     Expr
 }
 
 // UnaryExpr.Operator
 const (
-	UPlusStr  = '+'
-	UMinusStr = '-'
-	TildaStr  = '~'
+	UPlusStr  = "+"
+	UMinusStr = "-"
+	TildaStr  = "~"
+	BangStr   = "!"
+	BinaryStr = "binary"
 )
 
 // Format formats the node.
 func (node *UnaryExpr) Format(buf *TrackedBuffer) {
 	if _, unary := node.Expr.(*UnaryExpr); unary {
-		buf.Myprintf("%c %v", node.Operator, node.Expr)
+		buf.Myprintf("%s %v", node.Operator, node.Expr)
 		return
 	}
-	buf.Myprintf("%c%v", node.Operator, node.Expr)
+	if node.Operator == BinaryStr {
+		buf.Myprintf("%s %v", node.Operator, node.Expr)
```

Comment:
as it turns out, no.
if you, because you're silly, write "binary binary 'foo'" then to get the spacing right on that, as well as the normal case, we've got to do it this way.
if we make it "binary ", then the others have to be "+ " etc, and it looks wrong